### PR TITLE
Separating the URL construction methods: dotnet-install.sh

### PR DESCRIPTION
**Pull request by ghost**
Separating the 'legacy' URL construction from the 'current' URL construction methods and logic.

Why?
RedHat6 was not supported in the legacy time space, therefore the script should not attempt resolve to a RedHat6 distro specific URL when constructing the legacy download link. In that way, we will not infer that 1.0 supported RedHat6; et al. The 2 logical methods for URL construction should not share execution paths.

Addresses: https://github.com/dotnet/cli/issues/7752

### CODE REVIEW

File: scripts/obtain/dotnet-install.sh [hunk 1]
```diff
@@ -115,6 +111,30 @@ get_os_download_name_from_platform() {
...
 
+get_linux_platform_name() {
+    eval $invocation
+
+    if [ -n "$runtime_id" ]; then
+        echo "${runtime_id%-*}"
+        return 0
+    else
+        if [ -e /etc/os-release ]; then
+            . /etc/os-release
```
Comment (wli3) [inline scripts/obtain/dotnet-install.sh]: what is this for?

**Review by wli3 — APPROVED** [review]
so sorry for the delay

**Comment by wli3** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test RHEL6 x64 Debug Build

**Comment by wli3** [comment]
@dotnet-bot test RHEL6 x64 Debug Build

**Comment by wli3** [comment]
>  unable to prepare context: path "scripts/docker/rhel.6" not found
docker file is absent? Or it should be named rhel.6 but a different distro?

**Comment by johnbeisner** [comment]
closing...

**Pull request closed without merging**